app/lib/lib/roomTypes/newsfeed.js

```diff
@@ -0,0 +1,128 @@
+import { Meteor } from 'meteor/meteor';
+
+import { openRoom } from '../../../ui-utils';
+import { ChatRoom, ChatSubscription } from '../../../models';
+import { settings } from '../../../settings';
+import { hasAtLeastOnePermission } from '../../../authorization';
+import { getUserPreference, RoomTypeConfig, RoomTypeRouteConfig, RoomSettingsEnum, UiTextContext } from '../../../utils';
+import { getAvatarURL } from '../../../utils/lib/getAvatarURL';
+
+export class NewsfeedRoomRoute extends RoomTypeRouteConfig {
+	constructor() {
+		super({
+			name: 'newsfeed',
+			path: '/newsfeed/:name',
+		});
+	}
+
+	action(params) {
+		return openRoom('n', params.name);
+	}
+}
+
+export class NewsfeedRoomType extends RoomTypeConfig {
+	constructor() {
+		super({
+			identifier: 'n',
+			order: 30,
+			icon: 'star',
+			label: 'Newsfeed',
+			route: new NewsfeedRoomRoute(),
+		});
+	}
+
+	getIcon(roomData) {
+		if (roomData.prid) {
+			return 'discussion';
+		}
+		return this.icon;
+	}
+
+	findRoom(identifier) {
+		const query = {
+			t: 'n',
+			name: identifier,
+		};
+		return ChatRoom.findOne(query);
+	}
+
+	roomName(roomData) {
+		if (roomData.prid) {
+			return roomData.fname;
+		}
+		if (settings.get('UI_Allow_room_names_with_special_chars')) {
+			return roomData.fname || roomData.name;
+		}
+		return roomData.name;
+	}
+
+	condition() {
+		const groupByType = getUserPreference(Meteor.userId(), 'sidebarGroupByType');
+		return groupByType && (hasAtLeastOnePermission(['view-c-room', 'view-joined-room']) || settings.get('Accounts_AllowAnonymousRead') === true);
+	}
+
+	showJoinLink(roomId) {
+		return !!ChatRoom.findOne({ _id: roomId, t: 'n' });
+	}
+
+	includeInRoomSearch() {
+		return true;
+	}
+
+	isGroupChat() {
+		return true;
+	}
+
+	canAddUser(room) {
+		return hasAtLeastOnePermission(['add-user-to-any-c-room', 'add-user-to-joined-room'], room._id);
```

Comment:
Shouldn't canAddUser be always `false` for Newsfeed type room? What's the reason behind keeping it like this?

Reply:
I actually just based the newsfeed off private group template. That can be removed for now, but I think should be kept until I perfect the message relaying to make sure that we do not have to switch paths.